builtin/logical/aws/client.go

```diff
@@ -29,6 +30,16 @@ func getRootConfig(s logical.Storage) (*aws.Config, error) {
 		credsConfig.AccessKey = config.AccessKey
 		credsConfig.SecretKey = config.SecretKey
 		credsConfig.Region = config.Region
+		credsConfig.IAMEndpoint = config.IAMEndpoint
+		credsConfig.STSEndpoint = config.STSEndpoint
```

Comment:
This is the only place `credsConfig.IAMEndpoint` and `credsConfig.STSEndpoint` are referenced. I don't think you need them in the `CredentialsConfig` struct at all because, below, you're putting the endpoint directly in the `aws.Config` object.